lib/sinon/match.js

```diff
@@ -221,12 +221,50 @@ match.hasOwn = createPropertyMatcher(function (actual, property) {
     return actual.hasOwnProperty(property);
 }, "hasOwn");
 
+match.array = match.typeOf("array");
+
+match.array.deepEquals = function (expectation) {
+    return match(function (actual) {
+        // Comparing lengths is the fastest way to spot a difference before iterating through every item
+        var sameLength = actual.length === expectation.length;
+        return sameLength && Array.prototype.every.call(actual, function (element, index) {
```

Comment:
@fatso you're right!
I've just seen you doing that on the `behavior` module for `push` and it looks much better.
I think I'll start doing this too the next time I have to call `Array` methods like this. Thanks for the tip, I'm gonna update this PR as soon as I get home :smile: